src/Compilers/Core/Portable/SourceGeneration/GeneratorAdaptor.cs

```diff
@@ -24,25 +24,25 @@ public SourceGeneratorAdaptor(ISourceGenerator generator)
 
         public void Initialize(IncrementalGeneratorInitializationContext initContext)
         {
-            GeneratorInitializationContext generatorInitContext = new GeneratorInitializationContext(initContext.CancellationToken);
+            GeneratorInitializationContext generatorInitContext = new GeneratorInitializationContext(CancellationToken.None);
             SourceGenerator.Initialize(generatorInitContext);
 
             initContext.InfoBuilder.PostInitCallback = generatorInitContext.InfoBuilder.PostInitCallback;
             var syntaxContextReceiverCreator = generatorInitContext.InfoBuilder.SyntaxContextReceiverCreator;
 
             initContext.RegisterExecutionPipeline((executionContext) =>
             {
-                var contextBuilderSource = executionContext.Sources.Compilation
-                                            .Transform(c => new GeneratorContextBuilder(c))
-                                            .Join(executionContext.Sources.ParseOptions).Transform(p => p.Item1 with { ParseOptions = p.Item2.FirstOrDefault() })
-                                            .Join(executionContext.Sources.AnalyzerConfigOptions).Transform(p => p.Item1 with { ConfigOptions = p.Item2.FirstOrDefault() })
```

Comment:
Single value provider means no `.FirstOrDefault()` and compile time checked instead.